src/futils.cpp

```diff
@@ -502,4 +494,71 @@ namespace Exiv2 {
         const size_t idxLastSeparator = ret.find_last_of(EXV_SEPARATOR_STR);
         return ret.substr(0, idxLastSeparator);
     }
+
+    static bool pushPath(std::string& path,std::vector<std::string>& libs,std::set<std::string> & paths)
+    {
+        bool result = Exiv2::fileExists(path,true) && paths.find(path) == paths.end() && path != "/" ;
+        if ( result ) {
+            paths.insert(path);
+            libs.push_back(path);
+        }
+        return result ;
+    }
+
+    std::vector<std::string> getLoadedLibraries()
+    {
+        std::vector<std::string>  libs ;
+        std::set<std::string>     paths;
+        std::string               path ;
+
+    #if defined(WIN32) || defined(__CYGWIN__) || defined(__MINGW__)
+        // enumerate loaded libraries and determine path to executable
+        HMODULE handles[200];
+        DWORD   cbNeeded;
+        if ( EnumProcessModules(GetCurrentProcess(),handles,lengthof(handles),&cbNeeded)) {
+            char szFilename[_MAX_PATH];
+            for ( DWORD h = 0 ; h < cbNeeded/sizeof(handles[0]) ; h++ ) {
+                GetModuleFileNameA(handles[h],szFilename,lengthof(szFilename)) ;
```

Comment:
```suggestion
                GetModuleFileNameA(handles[h],szFilename,EXV_COUNTOF(szFilename)) ;
```

Reply:
You're right Dan.  That's what I should have done.  I added macro lengthof() to src/futils.cpp